test/cloudant.test.js

```diff
@@ -197,6 +200,93 @@ describe('cloudant connector', function() {
       });
   });
 
+  describe('model with array props gets updated properly', function() {
+    var prod1, prod2;
+    before('create Product', function(done) {
+      Product.create({
+        id: 1,
+        name: 'bread',
+        price: 100,
+        releases: [1, 2, 3],
+        type: ['plain', 'sesame', 'whole wheat'],
+        foo: [{id: 1, name: 'bread1'}, {id: 2, name: 'bread2'}],
+      }, function(err, product) {
+        if (err) return done(err);
+        prod1 = product;
+        Product.create({
+          id: 2,
+          name: 'bagel',
+          price: 100,
+          releases: [1, 2, 3],
+          type: ['plain', 'sesame', 'whole wheat'],
+          foo: [{id: 1, name: 'bagel1'}, {id: 2, name: 'bagel2'}],
+        }, function(err, product) {
+          if (err) return done(err);
+          prod2 = product;
+          done();
+        });
+      });
+    });
+
+    after(function(done) {
+      Product.destroyAll(function(err) {
+        if (err) return done(err);
+        done();
+      });
+    });
+
+    it('updates a single model with array props',
+      function(done) {
+        prod1.setAttribute('type', ['cinnamon raisin']);
+        prod1.setAttribute('releases', [4, 5, 6]);
+        prod1.setAttribute('foo', [{id: 3, name: 'bread3'}]);
+        Product.updateAll({id: 1}, prod1, {}, function(err, res) {
+          if (err) return done(err);
```

Comment:
@ssh24 could you take a look of https://github.com/strongloop/loopback-connector-cloudant/pull/115/files#r114569262 :)
Actually I think `if (err) return done(err)` would be better... IIRC, we spent huge effort in juggler to refactor the assertion from using should to using return done(err) :p

Copied from that comment:
> should.not.exist(err) and if (err) return done(err) does same check, and both skip other assertions if error is not null, the difference is the later one gives you a better error log.
When mocha handles the error, it consoles a well formatted error message, while should will display it as a failed assertion, expected, actual, many lines of +, -, etc...